Client/Frontend/Library/HistoryPanel/HistoryPanelViewModel.swift

```diff
@@ -266,16 +289,35 @@ class HistoryPanelViewModel: Loggable, FeatureFlaggable {
 
         isFetchInProgress = true
 
-        return profile.history.getSitesByLastVisit(limit: queryFetchLimit, offset: currentFetchOffset) >>== { result in
-            // Force 100ms delay between resolution of the last batch of results
-            // and the next time `fetchData()` can be called.
-            DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
-                self.isFetchInProgress = false
-
-                self.browserLog.debug("currentFetchOffset is: \(self.currentFetchOffset)")
+        switch profile.historyApiConfiguration {
+            case .old:
+            return profile.history.getSitesByLastVisit(limit: queryFetchLimit, offset: currentFetchOffset) >>== { result in
+                // Force 100ms delay between resolution of the last batch of results
+                // and the next time `fetchData()` can be called.
+                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
+                    self.isFetchInProgress = false
+                    self.browserLog.debug("currentFetchOffset is: \(self.currentFetchOffset)")
+                }
+                return deferMaybe(result)
+            }
+            case .new:
+            return profile.places.getVisitPageWithBound(
+                limit: queryFetchLimit,
+                offset: currentFetchOffset,
+                excludedTypes: VisitTransitionSet(0)
+            ) >>== { result in
+                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
+                    self.isFetchInProgress = false
+                    self.browserLog.debug("currentFetchOffset is: \(self.currentFetchOffset)")
+                }
+                let sites = result.infos.map { info in
+                    let title = (info.title ?? "").isEmpty ? info.url : info.title!
```

Comment:
We should force `info.title!` here, would be better to unwrap it.

Reply:
Forcing can cause unexpected behavior did you mean we `shouldn't` force? @lmarceau 😅

Reply:
Yes, missed a word there hahaha